plugin.video.viaplay/addon.py

```diff
@@ -3,601 +3,415 @@
-    except KeyError:
-        year = None
-    try:
-        genres = []
-        for genre in product['_links']['viaplay:genres']:
-            genres.append(genre['title'])
-        genre = ', '.join(genres)
-    except KeyError:
-        genre = None
-    try:
-        mpaa = product['content']['parentalRating']
-    except KeyError:
-        mpaa = None
-
-    if content == 'episode':
-        mediatype = 'episode'
-        title = product['content']['series']['episodeTitle'].encode('utf-8')
-        tvshowtitle = product['content']['series']['title'].encode('utf-8')
-        season = int(product['content']['series']['season']['seasonNumber'])
-        episode = int(product['content']['series']['episodeNumber'])
-        plot = product['content']['synopsis'].encode('utf-8')
-
-    elif content == 'series':
-        mediatype = 'tvshow'
-        title = product['content']['series']['title'].encode('utf-8')
-        tvshowtitle = product['content']['series']['title'].encode('utf-8')
-        try:
-            plot = product['content']['series']['synopsis'].encode('utf-8')
-        except KeyError:
-            plot = product['content']['synopsis'].encode('utf-8')  # needed for alphabetical listing
-
-    elif content == 'movie':
-        mediatype = 'movie'
-        title = product['content']['title'].encode('utf-8')
-        plot = product['content']['synopsis'].encode('utf-8')
-        try:
-            for actor in product['content']['people']['actors']:
-                cast.append(actor)
-        except KeyError:
-            pass
-        try:
-            directors = []
-            for director in product['content']['people']['directors']:
-                directors.append(director)
-            director = ', '.join(directors)
-        except KeyError:
-            pass
-
-    elif content == 'sport':
-        mediatype = 'video'
-        title = product['content']['title'].encode('utf-8')
-        plot = product['content']['synopsis'].encode('utf-8')
-
-    info = {
-        'mediatype': mediatype,
-        'title': title,
-        'tvshowtitle': tvshowtitle,
-        'season': season,
-        'episode': episode,
-        'year': year,
-        'plot': plot,
-        'duration': duration,
-        'code': imdb_code,
-        'rating': rating,
-        'votes': votes,
-        'genre': genre,
-        'director': director,
-        'mpaa': mpaa,
-        'cast': cast
+    for date in dates:
+        params = {
+            'action': 'list_products',
+            'url': date['href']
+        }
+        helper.add_item(date['date'], params)
+    helper.eod()
+
+
+def channels_page(url):
+    channels_dict = helper.vp.get_channels(url)
+
+    for channel in channels_dict['channels']:
+        params = {
+            'action': 'list_products',
+            'url': channel['_links']['self']['href']
+        }
+        art = {
+            'thumb': channel['content']['images']['fallback']['template'].split('{')[0],
```

Comment:
You could probably catch exceptions here. Some times there are no images returned by the REST API for some content and as a result your plugin will crash here when most likely it could still proceed to playback.

Reply:
@enen92 I've thought about it, but their API always supplies generic images if there are none available.

Reply:
Ah, don't worry about it then :)